### ISSUE TITLE
`Cannot invoke method name() on null object` trying to apply example from `README.md`

### ISSUE THREAD

**Issue by perlun** [etiennestuder/gradle-jooq-plugin#204]
Hi,

Thanks for an interesting plugin. Trying to convert my code-based jOOQ code gen example, but running into some interesting issues with trying out the Groovy DSL example in the repo README:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/per/git/some/project/build.gradle' line: 80

* What went wrong:
A problem occurred evaluating project ':some-project'.
> Cannot invoke method name() on null object

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

It seems to be these parts in the config that causes this:

```groovy
properties {
    property {
        key = 'ssl'
        value = 'true'
    }
}
```

If I remove the above block, I get a similar error on this block instead:

```groovy
forcedTypes {
    forcedType {
        name = 'varchar'
        includeExpression = '.*'
        includeTypes = 'JSONB?'
    }
    forcedType {
        name = 'varchar'
        includeExpression = '.*'
        includeTypes = 'INET'
    }
}
```

Could it be that these options are not supported with PostgreSQL or something? :thinking: (I noted that the example [here](https://github.com/etiennestuder/gradle-jooq-plugin/blob/master/example/use_groovy_dsl/build.gradle) seems to use the H2 JDBC driver.)

**Comment by lukaseder** [comment]
I vaguely remember there's some groovy syntax caveat with reserved names, see e.g.:

- https://stackoverflow.com/q/11388838/521799
- https://github.com/jOOQ/jOOQ/issues/4797
- https://groups.google.com/forum/#!topic/jooq-user/wi4S9rRxk4A

The workaround tends to be to prevent the ambiguity by adding `([:])` at some places, e.g. `property([:])`

This used to be documented in the manual, when it documented using the `groovy.xml.MarkupBuilder`:
https://web.archive.org/web/20200602093159/http://www.jooq.org/doc/latest/manual/code-generation/codegen-gradle/

We no longer document things this way, partially, because of precisely this error here. Instead, we recommend people use the programmatic API directly, without any such DSL magic that goes wrong here and there, see https://www.jooq.org/doc/latest/manual/code-generation/codegen-gradle/

**Comment by perlun** [comment]
@lukaseder Thanks for the valuable feedback here. :pray: It turned out that the reason this doesn't work (and the plugin does not support jOOQ 3.16) is most likely the changes introduced in https://github.com/jOOQ/jOOQ/issues/9641. The plugin makes blind assumptions on `javax.xml` annotations being present in the jOOQ codebase, which is no longer the case in `>= 3.16.0` versions where this has been replaced by `jakarta.xml`. I'm looking into it to see if we can come up with a fix, but it's a bit complex since our own codebase is still Java 8 only so we will also have to make the plugin work on Java 8 again (including not breaking any of our own usage of `javax.xml` in other parts of our codebase... so... :see_no_evil: It's complicated.)

I hear you re. using the programmatic API directly. The problem with doing it like that is that we lose some of the benefits of the Gradle build by doing it like that (build tasks which have source files, generated output files and so forth); we are planning to be able to run the codegen from DDL on the fly for each build and not keep the codegen-related files committed to our git repo. Hence, the plugin (if we can make it work, one way or another :sunglasses:) would give us quite a lot of value, or so we anticipate.

(We have a large, complex Gradle build with an estimate of about 80-90 subprojects, using Gradle's [multi-project build feature](https://docs.gradle.org/current/userguide/intro_multi_project_builds.html). Since we rely so heavily on Gradle already, trying to fit this as a new "piece" of the puzzle makes quite a bit of sense in our use case.)

**Comment by mrozanc** [comment]
@perlun I tested a change on top of your fix for the XSD path. Just compiling with the 2 APIs and implementations and chosing which one should be used according to which annotation jOOQ is using.

I ran the tests with jOOQ 3.15.1 then replacing it by v3.16.3 to see if it works.
JDK8 -> jooq 3.15.1 tests ok
JDK11 -> jooq 3.15.1 tests ok
JDK8 -> jooq 3.16.3 tests ok
JDK11 -> jooq 3.15.1 tests ok

I put this PR here: #208

**Comment by perlun** [comment]
For the reference, in case anyone is interested: the root cause for this was that it (the plugin) tried to look for a nonexistent `XmlElement` annotation (`javax.xml` instead of `jakarta.xml`), i.e. these lines: https://github.com/etiennestuder/gradle-jooq-plugin/blob/c69d3ee1dd7f179733ccc66e6867a76888ade77c/src/main/groovy/nu/studer/gradle/jooq/jaxb/JaxbConfigurationBridge.groovy#L65-L67

The recent fixes in this repo (#208 and/or https://github.com/etiennestuder/gradle-jooq-plugin/commit/1c84650c8e530a8155dc205a16d0c3681e5b6828) fixes support for `jakarta.xml`-based JAXB, so I'll close this for now. :+1: Older versions of the plugin (up to and including 6.0.1) **do not work** with jOOQ >= 3.16 because of this and some other issues.

**Comment by etiennestuder** [comment]
This is fixed in the recently relased v7.0 of the gradle-jooq-plugin. The problems was due to a backward-incompatible change in jOOQ between 3.15 and 3.16.
